src/20240603_Multi_MayFundingUpdate/AaveV3Polygon_MayFundingUpdate_20240603.sol

```diff
@@ -0,0 +1,149 @@
+
+interface IAavePolEthERC20Bridge {
+  function bridge(address token, uint256 amount) external;
+}
+
+interface IWMatic {
+  function withdraw(uint256 amount) external;
+}
+
+interface IAavePolEthPlasmaBridge {
+  function bridge(uint256 amount) external;
+}
+
+/**
+ * @title May Funding Update
+ * @author karpatkey_TokenLogic
+ * - Snapshot: Direct-to-AIP
+ * - Discussion: https://governance.aave.com/t/arfc-may-funding-update/17768
+ */
+contract AaveV3Polygon_MayFundingUpdate_20240603 is IProposalGenericExecutor {
+  using SafeERC20 for IERC20;
+
+  IAavePolEthERC20Bridge public constant BRIDGE =
+    IAavePolEthERC20Bridge(MiscPolygon.AAVE_POL_ETH_BRIDGE);
+
+  function execute() external {
+    _migrate();
+    _bridge();
+  }
+
+  function _migrate() internal {
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.USDT_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.USDT_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1e6
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.DAI_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.DAI_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1e18
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.WETH_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.WETH_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1e18
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV2PolygonAssets.USDT_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV2PolygonAssets.DAI_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV2PolygonAssets.WETH_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    uint256 usdtAmount = IERC20(AaveV2PolygonAssets.USDT_UNDERLYING).balanceOf(address(this));
+    IERC20(AaveV2PolygonAssets.USDT_UNDERLYING).forceApprove(
+      address(AaveV3Polygon.POOL),
+      usdtAmount
+    );
+
+    uint256 daiAmount = IERC20(AaveV2PolygonAssets.DAI_UNDERLYING).balanceOf(address(this));
+    IERC20(AaveV2PolygonAssets.DAI_UNDERLYING).forceApprove(address(AaveV3Polygon.POOL), daiAmount);
+
+    uint256 wethAmount = IERC20(AaveV2PolygonAssets.WETH_UNDERLYING).balanceOf(address(this));
+    IERC20(AaveV2PolygonAssets.WETH_UNDERLYING).forceApprove(
+      address(AaveV3Polygon.POOL),
+      wethAmount
+    );
+
+    AaveV3Polygon.POOL.deposit(
+      AaveV2PolygonAssets.USDT_UNDERLYING,
+      usdtAmount,
+      address(AaveV3Polygon.COLLECTOR),
+      0
+    );
+
+    AaveV3Polygon.POOL.deposit(
+      AaveV2PolygonAssets.DAI_UNDERLYING,
+      daiAmount,
+      address(AaveV3Polygon.COLLECTOR),
+      0
+    );
+
+    AaveV3Polygon.POOL.deposit(
+      AaveV2PolygonAssets.WETH_UNDERLYING,
+      wethAmount,
+      address(AaveV3Polygon.COLLECTOR),
+      0
+    );
+  }
+
+  function _bridge() internal {
```

Comment:
@defijesus We have two USDC's on polygon, wanted to double check if we wish to bridge the other USDC from the collector too